pkg/api/admin/openshiftcluster_validatestatic.go

```diff
@@ -0,0 +1,298 @@
+package admin
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"net"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/immutable"
+	"github.com/Azure/ARO-RP/pkg/util/subnet"
+)
+
+var (
+	rxResourceGroupID = regexp.MustCompile(`(?i)^/subscriptions/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/resourceGroups/[-a-z0-9_().]{0,89}[-a-z0-9_()]$`)
+	rxSubnetID        = regexp.MustCompile(`(?i)^/subscriptions/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/resourceGroups/[-a-z0-9_().]{0,89}[-a-z0-9_()]/providers/Microsoft\.Network/virtualNetworks/[-a-z0-9_.]{2,64}/subnets/[-a-z0-9_.]{2,80}$`)
+	rxDomainName      = regexp.MustCompile(`^` +
+		`([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9])` +
+		`(\.([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9]))*` +
+		`$`)
+)
+
+type openShiftClusterValidator struct {
+	sv openShiftClusterStaticValidator
+	dv openShiftClusterDynamicValidator
+}
+
+type openShiftClusterStaticValidator struct {
+	location   string
+	resourceID string
+
+	r azure.Resource
+}
+
+// Validate validates an OpenShift cluster
+func (v *openShiftClusterValidator) Static(_oc interface{}, _current *api.OpenShiftCluster) error {
+	if _current == nil {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeRequestNotAllowed, "", "Admin API does not allow cluster creation.")
```

Comment:
`Static(_oc, _current, isAdmin bool)`? or smth like this?

Reply:
Because the validator is a API-version specific - we know that we are in the `admin` or in `v20191231preview` version. So there is no need to pass `isAdmin` as an argument.